### PR TITLE
Bug/2761 no response if not all payload is read

### PR DESCRIPTION
**Pull request by kzangeli**
Checking HTTP header 'Content-Length' for **too long** message before even starting to read the message body, and returning error if the body is too long (according to Content-Length).

Fixes issue #2761

### CODE REVIEW

File: src/lib/rest/rest.cpp [hunk 1]
```diff
@@ -1220,6 +1220,18 @@ static int connectionTreat
     ciP->httpHeader.push_back("Fiware-Correlator");
     ciP->httpHeaderValue.push_back(ciP->httpHeaders.correlator);
 
+    if ((ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE) && (ciP->apiVersion == V2))
```
Comment (fgalan) [inline src/lib/rest/rest.cpp]: What about not NGSIv1 requests or out-of-NGSI requests?

Comment (kzangeli) [inline src/lib/rest/rest.cpp]: NGSI v1 requests are tricky.
Each type of request has a different response payload.
Not sure it is worth the effort to implement this for NGSI v1
Out-of-NGSI requests have very small if any payload at all.
So, not very interesting, could be implemented though. Easily.

Comment (fgalan) [inline src/lib/rest/rest.cpp]: It doesn't worth the effort.

NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**